sources/web/datalab/server.ts

```diff
@@ -175,6 +175,9 @@ function handleRequest(request: http.ServerRequest,
   //       this into a real feature, that involves a confirmation prompt, as
   //       well validation to require a POST request.
   if (path.indexOf('/_restart') == 0) {
+    if ('POST' != request.method) {
```

Comment:
Since this now requires a post request, we should also add a button in the app bar to send such a request (it's not reasonable to require the user to manually send one).